### PR TITLE
menu and pin animation

### LINKED ISSUE #3884: Select menu/quiz animation
https://satoshilabs.slack.com/archives/C06MEE8QH33/p1717428329817479

### LINKED ISSUE #3885: PIN keyboard animation
https://satoshilabs.slack.com/archives/C06MEE8QH33/p1717428329817479

### LINKED ISSUE #3896: Menu animation
https://satoshilabs.slack.com/archives/C06MEE8QH33/p1717512225458789

### PR DESCRIPTION
**Pull request by TychoVrahe**
Implements menu and pin animations.

On side, the backlight fading when changing layouts in (mostly) removed for mercury ui - in favor for transition animations. The homescreen/lockscreen attach animation is added that emulates the backlight fading effect, but the overlay animation is able to run simultaneously.

When animation is interrupted by a silent call etc, the animation is resumed where it stopped before - for this, new AttachType, AttachType::Resume, is introduced.

fixes #3884 fixes #3885 fixes #3896 

As part of the PIN animation, the rounded corners for corner buttons of the keyboard are added, hence the UI diffs. No other keyboards were modified this way.

### CODE REVIEW

**Comment by TychoVrahe** [comment]
rebased.

**Review by mmilata — APPROVED** [review]
LGTM, but only lightly tested.

This will conflict with #4024, please coordinate merging with @matejcik.

File: core/embed/rust/src/ui/model_mercury/cshape/keyboard_overlay.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            "Too small buffer"
+        );
+
+        self.prepare_overlay(&mut overlay_canvas);
+
+        canvas.blend_bitmap(bounds, overlay_canvas.view().with_fg(Color::black()));
+    }
+
+    fn cleanup(&mut self, _cache: &DrawingCache<'a>) {
+        // TODO: inform the cache that we won't use the zlib slot anymore
```
[the remark this answers is not in the document]
Comment (TychoVrahe) [reply core/embed/rust/src/ui/model_mercury/cshape/keyboard_overlay.rs]: no i believe its not needed, was copied from the other overlay. removed the comment: https://github.com/trezor/trezor-firmware/pull/3987/commits/3eb58567864f3ddf52327bb133b91bbd1d3757bb

File: core/embed/rust/src/ui/layout/obj.rs [hunk 2]
```diff
@@ -5,62 +5,94 @@ use core::{
...
+            Qstr::MP_QSTR_SWIPE_DOWN => 3u8,
+            Qstr::MP_QSTR_SWIPE_LEFT => 4u8,
+            Qstr::MP_QSTR_SWIPE_RIGHT => 5u8,
+            _ => return Err(Error::AttributeError(attr)),
+        };
+        unsafe { dest.write(value.into()) };
+        Ok(())
+    };
+    unsafe { util::try_or_raise(block) }
+}
```
Comment (matejcik) [inline core/embed/rust/src/ui/layout/obj.rs:92]:
```suggestion

static ATTACH_TYPE: Type = obj_type! {
    name: Qstr::MP_QSTR_AttachType,
    locals: &obj_dict!(obj_map! {
        Qstr::MP_QSTR_INITIAL => Obj::small_int(0u16),
        Qstr::MP_QSTR_RESUME => Obj::small_int(1u16),
        Qstr::MP_QSTR_SWIPE_UP => Obj::small_int(2u16),
        Qstr::MP_QSTR_SWIPE_DOWN => Obj::small_int(3u16),
        Qstr::MP_QSTR_SWIPE_LEFT => Obj::small_int(4u16),
        Qstr::MP_QSTR_SWIPE_RIGHT => Obj::small_int(5u16),
    }),
};
```

**Comment by bosomt** [comment]
**QA OK**

**Info:**
 - Suite version: desktop 24.7.3 (19fd64f5b2ab1bc265821e34d4c4b9c910864d0e)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/24.7.3 Chrome/118.0.5993.159 Electron/27.3.8 Safari/537.36
 - OS: MacIntel
 - Screen: 1512x982
 - Device: Trezor T3T1 2.8.1 regular (revision 33f5d4733065ac631d61ff458a5db32c81641124)
 - Transport: BridgeTransport 2.0.33

**Pull request merged**